lib/dlopen.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const binding = process.binding('dlopen');
+
+exports.dlopen = function dlopen(name) {
+  var lib = new Buffer(binding.sizeof_uv_lib_t);
```

Comment:
> Also this has basically been the recommended way of allocating memory for structs/etc. in node-ffi for years now, so I would hate for that to be an issue

This hinges on whether you're alright with V8 cleaning up the pointer, or if you'd like manual control?